### ISSUE TITLE
WooCommerce attribute hooks not fired with API

### ISSUE THREAD

**Issue by b3rend** [woocommerce/woocommerce#16454]
## Prerequisites

- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate
- [x] The issue still exists against the latest `master` branch of WooCommerce on Github
- [x] I have attempted to find the simplest possible steps to reproduce the issue
- [ ] I have included a failing test as a pull request (Optional)

The only search result I find on "the internet" is my own stackoverflow topic, I'm surprised I am the first having this problem.. I must be doing something wrong.

## Steps to reproduce the issue

1. Setup 2 WordPress installations with WooCommerce installed on both
2. Install api client on parent WebShop
3. Create hooks on the child webshop for woocommerce_attribute_added and woocommerce_attribute_deleted.
4. When an attribute it saved on the parent, push it with the api to the child. (using the same hooks)
5. Attribute is saved properly, hook does not fire on child.

I run it in vagrant with 1 parent and 3 childs, however It should be easy to reproduce like this too:

1. Install WordPress and WooCommerce
2. add action for woocommerce_attribute_added and either debug in the function or log something when its executed.
3. enable the api and generate the keys
4. Do a manual post request to create a new woocommerce attribute 
5. Now you will see the hook is not fired.

## Expected/actual behavior

I would expect the hook to be fired.

Okay, maybe a little explanation is required: I'm creating a plug-in which makes it possible to have one WooCommerce parent webshop with the ability to add webshops (with api keys) to the parent and select products you want to share on what webshop. Everything fine, though when I add an attribute on the parent and with the woocommerce_attribute_added hook (works) insert it in the child (also works) the hook on the child is not fired. I need this hook to save in the child database that this attribute is pulled from the parent webshop and thus they are not allowed to change/delete it locally.

I already did the same for products and terms (categories and tags included), so I expected the woocommerce_attribute_xxx hooks to fire normally. It might have something to do with the fact I use edit_term and create_term for example for terms (which are WordPress hooks afaik) and the hooks for woocommerce attributes are woocommerce hooks (obv). They are also stored in their separate table so there aren't any other hooks I could use afaik.

## Isolating the problem

- [x] This bug happens with only WooCommerce plugin active
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/) ???? I have a clean WordPress install with only WooCommerce and my own plug-in loaded.
- [x] I can reproduce this bug consistently using the steps above

### WordPress Environment ###

Home URL: http://parent.bdg
Site URL: http://parent.bdg
WC Version: 3.1.1
Log Directory Writable: ✔
WP Version: 4.8.1
WP Multisite: –
WP Memory Limit: 1 GB
WP Debug Mode: –
WP Cron: ✔
Language: en_US

### Server Environment ###

Server Info: Apache/2.4.10 (Debian)
PHP Version: 5.6.24-0+deb8u1
PHP Post Max Size: 1 GB
PHP Time Limit: 0
PHP Max Input Vars: 1000
cURL Version: 7.38.0
OpenSSL/1.0.1t

SUHOSIN Installed: –
Max Upload Size: 1 GB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.1.1
WC Database Prefix: wp_
woocommerce_sessions: ✔
woocommerce_api_keys: ✔
woocommerce_attribute_taxonomies: ✔
woocommerce_downloadable_product_permissions: ✔
woocommerce_order_items: ✔
woocommerce_order_itemmeta: ✔
woocommerce_tax_rates: ✔
woocommerce_tax_rate_locations: ✔
woocommerce_shipping_zones: ✔
woocommerce_shipping_zone_locations: ✔
woocommerce_shipping_zone_methods: ✔
woocommerce_payment_tokens: ✔
woocommerce_payment_tokenmeta: ✔
MaxMind GeoIP Database: ❌ The MaxMind GeoIP Database does not exist - Geolocation will not function. You can download and install it manually from http://dev.maxmind.com/geoip/legacy/geolite/ to the path: . Scroll down to "Downloads" and download the "Binary / gzip" file next to "GeoLite Country". Please remember to uncompress GeoIP.dat.gz and upload the GeoIP.dat file only.

### Security ###

Secure connection (HTTPS): ❌Your store is not using HTTPS. Learn more about HTTPS and SSL Certificates.
Hide errors from visitors: ✔

### Active Plugins (2) ###

Woocommerce Parent Koppeling: by Berend de Groot – 1.0
WooCommerce: by Automattic – 3.1.1

### Settings ###

API Enabled: ✔
Force SSL: –
Currency: EUR (€)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: afdasfsda (afdasfsda)
exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)

### WC Pages ###

Shop base: #4 - /shop/
Cart: #5 - /cart/
Checkout: #6 - /checkout/
My account: #7 - /my-account/

### Theme ###

Name: Twenty Seventeen
Version: 1.3
Author URL: https://wordpress.org/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –
`
</details>

**Comment by mikejolley** [comment]
`woocommerce_attribute_added` does not fire when you add an attribute term, only when the attribute's taxonomy is created for the first time. So if your stores share a DB, it will only ever fire once. Use regular WP term hooks to track attribute terms.

**Comment by mikejolley** [comment]
Also to add, 3.2 introduces a consistent `wc_create_attribute` helper which is used in all places for attribute creation. So that hook will fire regardless of what creates the new attribute taxonomy.

**Comment by b3rend** [comment]
I am not talking about an attribute term.. I am talking about the **attribute**. The hook fires when I add the attribute in the wp-admin of the website. But does not fire when the I would create the exact same attribute using the api. I am talking about the attribute which is being saved in the "wp_woocommerce_attribute_taxonomies" table. The hook fires every-time you add a new attribute in the wp-admin. I tested this a lot of times on the parent website when I needed to push new added attributes from the parent to the child.
